### ISSUE TITLE
Reproducing the same random outputs in forward and backward passes

### ISSUE THREAD

**Issue by ag1988** [RobinBruegger/RevTorch#3]
Thanks for the great implementation Robin and @lucidrains !! I just wanted to ask you if pytorch will use the same random seed during the forward pass of F and the second forward pass of F during backward pass of the reversible block?  If not, can you:
1) During **forward** pass of the reversible block :  sample random seeds seed_F and seed_G and set random seeds just before the forward passes of F and G using torch.manual_seed(seed). Then save these seeds to be used during backprop. 
2) During **backward** pass of the reversible block :  retrieve the saved seeds seed_F and seed_G from ctx and set them before the second forward passes of F and G to ensure the same outputs as during the original forward pass. 

I am asking this as, in the Reformer LSHAttention, one samples the hashes and it can be a problem if the same hashes are not reproduced during the backward pass.

Thanks in advance,
Ankit

**Comment by lucidrains** [comment]
@ag1988  oh yes, you are absolutely right. there is a problem here. I'll get a PR out by tonight for this

**Comment by lucidrains** [comment]
fixed with https://github.com/RobinBruegger/RevTorch/pull/4

**Comment by ag1988** [comment]
@lucidrains Thanks! Looks good :+1:

**Comment by lucidrains** [comment]
@ag1988 I've temporarily updated reformer_pytorch package with the revtorch fix on 0.8.1 thanks so much for finding this bug!

**Comment by ag1988** [comment]
@lucidrains You're welcome Phil! Will be definitely be playing around with your reformer code in the coming weeks :smile_cat:.

**Comment by RobinBruegger** [comment]
@ag1988 @lucidrains  Added an option to use the same random seeds in forward and backward pass in Version 0.2.2 by merging [this PR](https://github.com/RobinBruegger/RevTorch/pull/4)
